SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityDeployChecker.java

```diff
@@ -370,74 +386,211 @@ private SingularityDeployResult getDeployResult(final SingularityRequest request
       return new SingularityDeployResult(DeployState.FAILED, String.format("Task(s) %s for this deploy failed", inactiveDeployMatchingTasks));
     }
 
-    if (shouldCheckLbState(pendingDeploy)) {
-      final SingularityLoadBalancerUpdate lbUpdate = lbClient.getState(getLoadBalancerRequestId(pendingDeploy.getDeployMarker()));
-
-      updateLoadBalancerStateForTasks(deployActiveTasks, LoadBalancerRequestType.ADD, lbUpdate);
-
-      DeployState deployState = interpretLoadBalancerState(lbUpdate, pendingDeploy.getCurrentDeployState());
-
-      updatePendingDeploy(pendingDeploy, lbUpdate, deployState);
-
-      if (deployState != DeployState.WAITING) {
-        return fromLbState(deployState, lbUpdate);
-      }
+    if (pendingDeploy.getDeployProgress().isPresent()) {
+      return checkDeployProgress(request, cancelRequest, pendingDeploy, updatePendingDeployRequest, deploy, deployActiveTasks, otherActiveTasks);
+    } else {
+      return new SingularityDeployResult(DeployState.FAILED, "No deploy progress present for deploy of long running request");
```

Comment:
This means that any pending deploy(s) when this change is rolled out will fail. It might be worth making a ZK migration for this (but I can also see the argument where that'd probably be overkill). A compromise might be to make this error message a little more specific (since folks would only hit it in this edge case, right?) -- maybe something like `No deploy progress data present in ZooKeeper. Please reattempt your deploy.`?